src/Core/Hook/Generator/HookDescriptionGenerator.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Hook\Generator;
+
+use PrestaShop\PrestaShop\Core\Hook\HookDescription;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifier;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use PrestaShop\PrestaShop\Core\Util\String\StringValidator;
+use PrestaShop\PrestaShop\Core\Util\String\StringValidatorInterface;
+
+/**
+ * Generates description for hook names.
+ */
+final class HookDescriptionGenerator implements HookDescriptionGeneratorInterface
```

Comment:
I'm wondering if it's worth it to somehow flag as a Dev Tool in order to remove it from the final PrestaShop zip archive 🤔 .

Reply:
not sure yet how to achieve it yet but not looking from the performance side this would be quite useful in my opinion -  a good separation of concerns :+1:

Reply:
We can add the tags later, the beta awaits us 😄

@eternoendless please have a look at that, tell us if it's worth it and easy to implement